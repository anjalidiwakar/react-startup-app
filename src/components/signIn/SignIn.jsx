
import React, { useState } from "react";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import SignUp from '../SignUp/Signup';
import "./SignIn.css";
import { Link } from 'react-router-dom';
import Admin from '../admin/Admin';

export default function Login(props) {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [renderState, setRenderState] = useState("");

    function validateForm() {
        return email.length > 0 && password.length > 0;
    }
    function handleSubmit() {
        let validationError = ""
        let users = localStorage.getItem("users");
        if (users) {
            users = JSON.parse(users);
        }
        let user = users.find(t => t.email === email);
        if (user) {
            if (user.password === password) {
                setRenderState("SignedIn");
                sessionStorage.setItem("email", email);
                sessionStorage.setItem("password", password);
                sessionStorage.setItem("role", user.role);
                props.setUserToken(true);
               // window.location.href = "http://localhost:3000/" + user.role;
            }
            else {
                validationError = "Password is incorrect, please try again.";
            }
        }
        else
            validationError = "No user found with this email, plrease try with another one";

    }
    const handleSignUp = () => { setRenderState("SignUp") }

    if (renderState === "") {
        return (
            <>
            <div className="login-wrapper">
                <p>Please Sign In</p>
                <div class="field">
                    <label class="label is-large">Email</label>
                    <div class="control">
                        <input class="input is-normal" type="email" value={email}
                            onChange={(e) => setEmail(e.target.value)} placeholder="Enter email" />
                    </div>
                </div>
                <div class="field">
                    <label class="label is-normal">Password</label>
                    <div class="control">
                        <input class="input is-normal" type="password" value={password}
                            onChange={(e) => setPassword(e.target.value)} placeholder="Enter password" />
                    </div>
                </div>

                <Button className="buttons button is-primary" onClick={handleSubmit} type="submit" disabled={!validateForm()}>
                    Login
            </Button>
                <p> Do not have an account? Please Sign Up here</p>
                <Button className="buttons button is-primary" onClick={handleSignUp}>SignUp</Button>
                
            </div>
            
    </>
        );
    }
    else if (renderState === "SignUp") {
        return (
            <div>
                <SignUp />
            </div>
        );
    }
}