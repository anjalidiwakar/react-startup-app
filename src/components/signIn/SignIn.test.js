import SignIn from './SignIn';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
const mockStore = configureStore([]);

describe("connected reatc-redux", () => {
    let store, component;
    beforeEach(() => {
        store = mockStore({
    });
    store.dispatch = jest.fn();
    component = renderer.create(
        <Provider store={store}>
          <SignIn />
        </Provider>
      );
  });
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
 
   
})