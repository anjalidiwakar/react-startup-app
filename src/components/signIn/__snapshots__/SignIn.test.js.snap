// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected reatc-redux  1`] = `
<div
  className="login-wrapper"
>
  <p>
    Please Sign In
  </p>
  <div
    className="field"
  >
    <label
      className="label is-normal"
    >
      Email
    </label>
    <div
      className="control"
    >
      <input
        className="input is-normal"
        onChange={[Function]}
        placeholder="Enter email"
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label is-normal"
    >
      Password
    </label>
    <div
      className="control"
    >
      <input
        className="input is-normal"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    className="button is-info btn btn-primary"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Sign in
  </button>
  <p>
     Do not have an account? Please Sign Up here
  </p>
  <button
    className="button is-info btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Sign Up
  </button>
</div>
`;

exports[`connected reatc-redux should render with given state from Redux store 1`] = `
<div
  className="login-wrapper"
>
  <p>
    Please Sign In
  </p>
  <div
    className="field"
  >
    <label
      className="label is-normal"
    >
      Email
    </label>
    <div
      className="control"
    >
      <input
        className="input is-normal"
        onChange={[Function]}
        placeholder="Enter email"
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label is-normal"
    >
      Password
    </label>
    <div
      className="control"
    >
      <input
        className="input is-normal"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    className="button is-info btn btn-primary"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Sign in
  </button>
  <p>
     Do not have an account? Please Sign Up here
  </p>
  <button
    className="button is-info btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Sign Up
  </button>
</div>
`;
